server/src/test/java/org/elasticsearch/index/mapper/RootObjectMapperTests.java

```diff
@@ -217,16 +217,14 @@ public void testRuntimeSectionMerge() throws IOException {
             merge(mapperService, mapping);
             DoubleScriptFieldType field = (DoubleScriptFieldType)mapperService.fieldType("field");
             assertEquals(NumberFieldMapper.NumberType.DOUBLE.typeName(), field.typeName());
-            LongScriptFieldType field2Updated = (LongScriptFieldType)mapperService.fieldType("field2");
-            assertSame(field2, field2Updated);
+            assertThat(mapperService.fieldType("field2"), instanceOf(LongScriptFieldType.class));
```

Comment:
so, now the runtime field is the same instance but not the mapped field type, or?

Reply:
yes, exactly.

Reply:
which makes me wonder: do we need to return a new instance each time asMappedFieldType is called? Probably not?

Reply:
We probably could create fixed MappedFieldTypes at construction time and always return the same object, but this is only called when Mappings are re-parsed so I'm not sure it really makes a lot of difference.

Reply:
it certainly complicates the test no? What do we do for field mappers? Don't we always expose the same field type, that is created at construction time?

Reply:
Have pushed a change to built the field types up front.